### PR TITLE
[Raft] Setup required states.

### PR DESCRIPTION
**Pull request by liana-yang**
image
image

### CODE REVIEW

**Review by flyingmeat — APPROVED** [review]

File: lab/src/raft/raft.go [hunk 1]
```diff
@@ -55,6 +55,21 @@ type Raft struct {
 	// Look at the paper's Figure 2 for a description of what
 	// state a Raft server must maintain.
 
+	// Persistent state on all servers
+	currentTerm int
+	votedFor    int
+	log         []LogEntry
```
Comment (flyingmeat) [inline lab/src/raft/raft.go:75]: Define LogEntry?

File: lab/src/raft/raft.go [hunk 2]
```diff
@@ -55,6 +55,21 @@ type Raft struct {
 	// Look at the paper's Figure 2 for a description of what
 	// state a Raft server must maintain.
 
+	// Persistent state on all servers
```
Comment (flyingmeat) [inline lab/src/raft/raft.go:72]: And there should be a status for server

File: lab/src/raft/raft.go (hunk 1 above)
Comment (liana-yang) [reply lab/src/raft/raft.go:75]: Done.

File: lab/src/raft/raft.go (hunk 2 above)
Comment (liana-yang) [reply lab/src/raft/raft.go:72]: Done.

**Pull request merged**